include/RAJA/policy/openmp_target/reduce.hpp

```diff
@@ -185,6 +193,13 @@ struct TargetReduce
   {
   }
 
+  void reset(T in_val, T identity_ = Reducer::identity())
+  {
+    operator T();
+    initVal = in_val;
+    finalVal = identity_;
```

Comment:
This isn't really something for this PR, but why does finalVal exist in this class at all?

Reply:
I don't see why identity is needded here if its not in the constructor.

Reply:
This should reset the ReduceData as well.

Reply:
Just pushed some updates, thanks!

Reply:
I'm not sure about finalVal, but I added the identity to keep it consistent with other back-ends.